script/oltpbenchmark/website/run.py

```diff
@@ -0,0 +1,198 @@
+    # 'twitter',
+    # 'voter',
+    # 'wikipedia',
+    'ycsb'
+]
+
+
+
+def config_xml_file(host, port, benchmark,
+                    transaction, scalefactor, terminals,
+                    time, upload_code, upload_url,
+                    config_in, config_out, username, password):
+    config_path = os.path.join(config_in, 'sample_{}_config.xml'.format(benchmark))
+    xml = ElementTree.parse(config_path)
+    root = xml.getroot()
+    root.find("dbtype").text = "postgres"
+    root.find("driver").text = "org.postgresql.Driver"
+    root.find("DBUrl").text = "jdbc:postgresql://{0}:{1}/{2}".format(host, port, benchmark)
+    root.find("username").text = username
+    root.find("password").text = password
+    uploadCodeElem = Element("uploadCode")
+    uploadCodeElem.text = upload_code.decode('utf-8')
+    root.append(uploadCodeElem)
+    uploadUrlElem = Element("uploadUrl")
+    uploadUrlElem.text = upload_url.decode('utf-8')
+    root.append(uploadUrlElem)
+    if 'isolation' not in root:
+        isoElem = Element("isolation")
+        isoElem.text = str(transaction)
+        root.append(isoElem)
+    root.find("isolation").text = str(transaction)
+    root.find("scalefactor").text = str(scalefactor)
+    root.find("terminals").text = str(terminals)
+    for work in root.find('works').findall('work'):
+        if 'time' not in work:
+            timeElem = Element("time")
+            timeElem.text = str(time)
+            work.append(timeElem)
+        work.find('time').text = str(time)
+        work.find('rate').text = "unlimited"
+    outfile = config_out + benchmark + '_config.xml'
+    xml.write(outfile, encoding='UTF-8')
+    print("Wrote {}".format(outfile))
+
+
+def run_commands(commands):
+    for command in commands:
+        x = command.split(' ')
+        if x[0] == 'cd':
+            os.chdir(x[1])
+        else:
+            os.system(command)
+
+
+def peloton_build():
+    peloton_path = PELOTON_BASE_DIR
+    install_script_path = os.path.join(peloton_path, 'script', 'installation')
+    build_path = os.path.join(peloton_path, 'build')
+    bin_path = os.path.join(peloton_path, 'bin')
+
+    commands = [
+        'cd ' + peloton_path,
+        'git pull',
+        'cd ' + install_script_path,
+        'bash packages.sh',
+        'cd ' + peloton_path,
+        'mkdir -p build',
+        'cd ' + build_path,
+        'cmake -DCMAKE_BUILD_TYPE=Release ..',
+        'make -j4',
+        'cd ' + install_script_path,
+        'bash create_certificates.sh',
+        'export PATH=' + bin_path + ":$PATH",
+        'cd ' + PELOTON_BASE_DIR,
+    ]
+
+    run_commands(commands)
+
+
+def peloton_init():
+    peloton_binary = os.path.join(PELOTON_BASE_DIR, 'build', 'bin', 'peloton')
+    script_dir = SCRIPT_DIR
+
+    commands = [
+        'cd ' + SCRIPT_DIR,
+        'killall peloton',
+        peloton_binary + ' -port 15721 &',
+        'sleep 3',
+        'psql -U postgres -h localhost -p 15721 default_database < init.sql',
```

Comment:
1. As per above comment, don't use init.sql. Construct the necessary sql commands and pass in here via the -c argument to psql.  

2. There are several places where the 15721 port number is hard coded in this script. It should not be. It should be a command line parameter, with a default value of 15721 if desired.  Pass it down through the layers.

3. Sleep as a synchronization mechanism works poorly and is best avoided unless there is no viable alternative. Here, It recommend lifting or adapting _wait_for_peloton from script/testing/junit/run_junit.py

4. killall is also best used only as a last resort. If you chose to run multiple copies of peloton, this would indiscriminately terminate all of them. Run peloton using subprocess, save the pid, then kill your process only. Again, see run_junit.py for code that can be re-used.